From c730e13d64dea13a3e965bbce094755bb4aad1a0 Mon Sep 17 00:00:00 2001
From: Wen He <wen.he_1@nxp.com>
Date: Thu, 29 Aug 2019 16:54:49 +0800
Subject: [PATCH 050/245] clk: ls1028a: Fixes interface compatilbility issue
 due the backport

commit e64401bb39b76296f98ac6aa57f3be1ee7309e42 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux.git

Use older interface to compatible older kernel LTS release.

Signed-off-by: Wen He <wen.he_1@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/clk/clk-plldig.c | 15 +++++----------
 1 file changed, 5 insertions(+), 10 deletions(-)

diff --git a/drivers/clk/clk-plldig.c b/drivers/clk/clk-plldig.c
index b8c3d420a5f0..015917cbfc38 100644
--- a/drivers/clk/clk-plldig.c
+++ b/drivers/clk/clk-plldig.c
@@ -203,7 +203,7 @@ static int plldig_clk_probe(struct platform_device *pdev)
 	struct resource *mem;
 	struct clk_init_data init = {};
 	struct device *dev = &pdev->dev;
-	struct clk_parent_data parent_data;
+	const char *parent_name;
 	int ret;
 
 	data = devm_kzalloc(dev, sizeof(*data), GFP_KERNEL);
@@ -215,12 +215,10 @@ static int plldig_clk_probe(struct platform_device *pdev)
 	if (IS_ERR(data->regs))
 		return PTR_ERR(data->regs);
 
-	parent_data.name = of_clk_get_parent_name(dev->of_node, 0);
-	parent_data.index = 0;
-
 	init.name = dev->of_node->name;
 	init.ops = &plldig_clk_ops;
-	init.parent_data = &parent_data;
+	parent_name = of_clk_get_parent_name(dev->of_node, 0);
+	init.parent_names = &parent_name;
 	init.num_parents = 1;
 
 	data->hw.init = &init;
@@ -232,11 +230,8 @@ static int plldig_clk_probe(struct platform_device *pdev)
 		return ret;
 	}
 
-	ret = devm_of_clk_add_hw_provider(dev, of_clk_hw_simple_get, &data->hw);
-	if (ret)
-		dev_err(dev, "failed adding the clock provider\n");
-
-	return ret;
+	return of_clk_add_hw_provider(dev->of_node, of_clk_hw_simple_get,
+								&data->hw);
 }
 
 static int plldig_clk_remove(struct platform_device *pdev)
-- 
2.17.1

